lib/rubygems/package.rb

```diff
@@ -239,12 +239,14 @@ def add_metadata tar # :nodoc:
   ##
   # Builds this package based on the specification set by #spec=
 
-  def build skip_validation = false
+  def build skip_validation = false, allow_warnings = true
```

Comment:
`package.build spec, true, false` is not a valid combination of arguments, checking it here seems best.